cluster/network.go

```diff
@@ -293,37 +355,45 @@ func (c *Cluster) deployTCPPortListeners(ctx context.Context) error {
 			"1337/tcp": {},
 		},
 	}
-
 	hostCfg := &container.HostConfig{
 		PortBindings: nat.PortMap{
 			"1337/tcp": portBindingList,
 		},
 	}
 
-	uniqHosts := c.getUniqueHostList()
-	for _, host := range uniqHosts {
+	return docker.DoRunContainer(ctx, host.DClient, imageCfg, hostCfg, containerName, host.Address, "network")
+}
+func (c *Cluster) removeTCPPortListeners(ctx context.Context) error {
+	log.Infof(ctx, "[network] Removing port listener containers")
+	var errgrp errgroup.Group
+
+	for _, host := range c.EtcdHosts {
 		runHost := host
 		errgrp.Go(func() error {
-			return docker.DoRunContainer(ctx, runHost.DClient, imageCfg, hostCfg, PortListenContainer, runHost.Address, "network")
+			return docker.DoRemoveContainer(ctx, runHost.DClient, EtcdPortListenContainer, runHost.Address)
 		})
 	}
+
 	if err := errgrp.Wait(); err != nil {
 		return err
 	}
-	log.Infof(ctx, "[network] Port listener containers deployed successfully")
-	return nil
-}
-func (c *Cluster) removeTCPPortListeners(ctx context.Context) error {
-	log.Infof(ctx, "[network] Removing port listener containers")
-	var errgrp errgroup.Group
+	for _, host := range c.ControlPlaneHosts {
```

Comment:
Same for remove. Lets think about extracting similar code, and passing hosts/container to remove as an argument.